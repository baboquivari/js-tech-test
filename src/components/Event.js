import React from 'react';
import { Link } from 'react-router-dom';

const Event = props => {
    const { eventName, eventId, primaryMarket, outcomes, showDecimalOdds } = props;

    const marketId = primaryMarket[0].marketId;
    const marketOutcomes = outcomes[marketId];
    // console.log(marketOutcomes);
    // console.log(primaryMarket[0]);

    return (
        <Link to={`/event/${eventId}`}>
            <li>{ eventName }</li>
            <div className="primaryMarket">
                <div id="primaryMarketName">
                    { primaryMarket[0].name }
                    ||
                </div>
                <div id="primaryMarketDescription">
                    { primaryMarket[0].type }
                </div>
                <div id="primaryMarketOutcomes">
                    { 
                        <ul>
                            { getPrices(marketOutcomes, showDecimalOdds) } 
                        </ul>
                    }
                </div>
            </div>
        </Link>
    );

    function getPrices (outcomes, showDecimalOdds) {
        return outcomes.map((outcome, i) => {
            return (
                <li key={i}>
                    { 
                        showDecimalOdds
                        ? outcome.price.decimal  
                        : `${outcome.price.num} / ${outcome.price.den}`
                    }
                </li>
            );
        });
    }
};

export default Event;