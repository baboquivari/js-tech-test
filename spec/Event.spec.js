import Event from '../src/components/Event';
import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';
import toJson from 'enzyme-to-json';

configure({ adapter: new Adapter() });

// snapshot testing
describe('Event', () => {
    it('should render correctly', () => {

		const output = shallow(
			<Event 
				eventName={'test'} 
				eventId={'0000'}
				primaryMarket={[
					{
						eventId: 8989898,
						marketId: 989898,
						name: 'Test',
						type: 'Football'
					}
					]}
				outcomes={{
					989898: [
						{
							eventId: 8484848,
							marketId: 8484333,
							name: 'Testing',
							price: {
								decimal: 1.3,
								den: 6,
								num: 2
							}
						}
					]
				}}
				showDecimalOdds={false}
			/>
		);
      
		expect(toJson(output)).toMatchSnapshot();
    });
});

// TODO: Add more unit tests. Also, how does my app respond to bad API responses?