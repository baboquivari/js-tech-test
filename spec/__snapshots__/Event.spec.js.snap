// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Event should render correctly 1`] = `
<Link
  replace={false}
  to="/event/0000"
>
  <li>
    test
  </li>
  <div
    className="primaryMarket"
  >
    <div
      id="primaryMarketName"
    >
      Test
      ||
    </div>
    <div
      id="primaryMarketDescription"
    >
      Football
    </div>
    <div
      id="primaryMarketOutcomes"
    >
      <ul>
        <li
          key="0"
        >
          2/6
        </li>
      </ul>
    </div>
  </div>
</Link>
`;
